### ISSUE TITLE
JLBH: [Feature] Use JLBH in xUnit types of tests

### ISSUE THREAD

**Issue by michaelszymczak** [OpenHFT/Chronicle-Core#45]
JLBH is a great and simple tool to measure latencies, partially thanks to easy to follow, human readable reports. Recently I tried to use it to use it as part of TDD process when developing components for a latency sensitive application and I realized that there is no API that can be used to create automated tests that can be run in a production-like, performance focused CI environment to avoid introducing latency regressions and to continuously increase performance of the existing app. The fully automated, xUnit style of testing would allow that. Example:
```java

  @Test
  public void shouldRespondWithin50usForTheMostCommonTypeOfRequest() throws Exception {
    // given
    JLBHTask unitUnderTest = new TaskInvokingTheUnitYouWantToMeasure();
    JLBH jlbh = new JLBH(new JLBHOptions().jlbhTask(unitUnderTest) /* other options */);

    // when
    jlbh.start();

    // then
    Result result = ... // somehow retrieve the result
    assertTrue(result.getEndToEndLatencies().get99thPercentileInNs() <= 50_000)
  }
```
If we had such tool, it could become a great tool for not only measuring, but also test driving latency sensitive applications.

**Comment by michaelszymczak** [comment]
Do you think it's maybe better to pass additional parameters via JLBHOptions instead of passing them directly to the constructor? #

**Comment by RobAustin** [comment]
you can get the `results` from your class that implements net.openhft.chronicle.core.jlbh.JLBHTask via the `void init(JLBH jlbh);`


```
public interface JLBHTask {
    /**
     * This method is called before the benchmark is started.
     * It gives the task a chance to do initialisation.
     *
     * @param jlbh A reference to the JLBH which is needed so that <code>jlbh.sample()</>
     *             can be invoked when the benchmark is complete. It can also be used to
     *             create more probes into the benchamrk.
     */
    void init(JLBH jlbh);

    /**
     * This method is called for each iteration of the benchmark.
     * The timestamp passed into the method is not the same as <code>System.nanoTime()</>.
     * It is the calculated time that the test is supposed to have started.
     *
     * @param startTimeNS The time that should be used as the start time for the sample.
     */
    void run(long startTimeNS);

    /**
     * Notify that warmup phase is over
     */
    default void warmedUp() {
    }

    /**
     * This method is used for any clean up that might be required by the benchmark.
     */
    default void complete() {
    }
}
````

```
result = jlbh.percentileRuns()
```

**Comment by michaelszymczak** [comment]
Thanks, It's good to know that JLBH now provides access to the raw percentiles. I think that obtaining them as presented https://github.com/OpenHFT/Chronicle-Core/blob/942f845184e433f59285a11e030f4004c29e84a6/src/test/java/net/openhft/chronicle/core/jlbh/JLBHTest.java will be still my preferred way as it

* provides better API (reduces Connascence of Meaning by providing the names of the percentiles as opposed to simply array of numbers)
* is thread safe, when you run the benchmark using multiple threads
* is unit tested

Either way, this issue can now be done, thanks for closing.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.17.1](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.17.1), [BOM-2.17.68](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.68)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.17.1](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.17.1), [BOM-2.17.72](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.72)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.17.1](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.17.1), [BOM-2.17.74](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.74)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.20.102](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.20.102), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
